nodejs/eks/cluster.ts

```diff
@@ -192,12 +193,17 @@ export function createCore(name: string, args: ClusterOptions, parent: pulumi.Co
 
     // Add any requested StorageClasses.
     const storageClasses = args.storageClasses || {};
+    const userStorageClasses: UserStorageClasses = pulumi.output({});
     if (typeof storageClasses === "string") {
         const storageClass = { type: storageClasses, default: true };
-        createStorageClass(`${name.toLowerCase()}-${storageClasses}`, storageClass, { parent: parent, provider: provider });
+        userStorageClasses.apply(usc => {
+            usc[storageClasses] = createStorageClass(`${name.toLowerCase()}-${storageClasses}`, storageClass, { parent: parent, provider: provider });
+        });
```

Comment:
I would not populate these asynchronously. There could be very subtle races here. 

The suggestions to make the output an Output was to wrap in `Pulumi.output` before storing into the return value to this function.

Reply:
Done